Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -46,12 +50,47 @@ internal struct BinaryDecoder: Decoder {
 
     private var complete: Bool {return available == 0}
 
-    internal init(forReadingFrom pointer: UnsafePointer<UInt8>, count: Int, extensions: ExtensionMap? = nil) {
+    internal init(
+      forReadingFrom pointer: UnsafePointer<UInt8>,
+      count: Int,
+      options: BinaryDecodingOptions,
+      extensions: ExtensionMap? = nil
+    ) {
         // Assuming baseAddress is not nil.
         p = pointer
         available = count
         fieldStartP = p
         self.extensions = extensions
+        self.options = options
+        recursionBudget = options.messageDepthLimit
+    }
+
+    internal init(
+      forReadingFrom pointer: UnsafePointer<UInt8>,
+      count: Int,
+      parent: BinaryDecoder
```

Comment:
Those differences look like noise, probably. I'd say go ahead and submit this since there doesn't seem to be a significant impact.

As for the modification, all the checkouts should be happening in a temp location but maybe there's an error in the script somewhere...